### ISSUE TITLE
E1041 false positive: Lambda Alias Ref rejected where Lambda Function ARN is expected

### ISSUE THREAD

**Issue by roger-zhangg** [aws-cloudformation/cfn-lint#4468]
### CloudFormation Lint Version

1.49.1

### What operating system are you using?

Ubuntu (GitHub Actions CI), also reproduced on Mac

### Describe the bug

**Describe the bug:**

After cfn-lint 1.49.0 expanded format keyword coverage (#4442), rule E1041 produces false positives when `{"Ref": "MyAlias"}` (an `AWS::Lambda::Alias` resource) is used in properties that expect `AWS::Lambda::Function.Arn` format.

Full error messages from the reproduction template:

```
template.yaml:34:15:34:28:E1041:{'Ref': 'MyAlias'} does not match destination format of 'AWS::Lambda::Function.Arn'
template.yaml:40:9:40:22:E1041:{'Ref': 'MyAlias'} does not match destination format of 'AWS::Lambda::Function.Arn'
```

This is a false positive. `Ref` on an `AWS::Lambda::Alias` returns the alias ARN, e.g. `arn:aws:lambda:us-east-1:123456789012:function:my-func:Live`. This is a qualified Lambda function ARN and is accepted by every AWS service wherever a Lambda function ARN is expected. This is the standard pattern used by the [AWS SAM transform](https://github.com/aws/serverless-application-model) to wire event sources to a Lambda alias instead of `$LATEST`.

A secondary case in the same category: `{"Ref": "MyKmsKey"}` (`AWS::KMS::Key`) is flagged in `KmsKeyArn` properties with:

```
template.yaml:10:9:10:20:E1041:{'Ref': 'myKey'} with formats ['AWS::KMS::Key.Id'] does not match destination format of 'AWS::KMS::Key.Arn'
```

While `Ref` on a KMS Key returns a Key ID (not an ARN), the Lambda `KmsKeyArn` API property accepts both Key ARNs and Key IDs at the API level.

This causes 12 E1041 errors in the SAM transform CI after upgrading to cfn-lint >= 1.49.0.

### Expected behavior

No error. `AWS::Lambda::Alias` `Ref` return value should be treated as compatible with `AWS::Lambda::Function.Arn` format, since an alias ARN is a qualified function ARN.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  MyRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Version: "2012-10-17"
        Statement:
          - Effect: Allow
            Principal:
              Service: lambda.amazonaws.com
            Action: sts:AssumeRole

  MyFunction:
    Type: AWS::Lambda::Function
    Properties:
      FunctionName: my-func
      Runtime: python3.13
      Handler: index.handler
      Code:
        ZipFile: "def handler(event, context): pass"
      Role: !GetAtt MyRole.Arn

  MyVersion:
    Type: AWS::Lambda::Version
    Properties:
      FunctionName: !Ref MyFunction

  MyAlias:
    Type: AWS::Lambda::Alias
    Properties:
      FunctionName: !Ref MyFunction
      FunctionVersion: !GetAtt MyVersion.Version
      Name: Live

  # E1041 false positive: Ref to Alias is a valid Lambda function ARN
  MyEventRule:
    Type: AWS::Events::Rule
    Properties:
      ScheduleExpression: "rate(1 hour)"
      Targets:
        - Id: Target1
          Arn: !Ref MyAlias

  # E1041 false positive: Ref to Alias is valid for FunctionName
  MyPermission:
    Type: AWS::Lambda::Permission
    Properties:
      Action: lambda:InvokeFunction
      FunctionName: !Ref MyAlias
      Principal: events.amazonaws.com
```

Output:
```
template.yaml:34:15:34:28:E1041:{'Ref': 'MyAlias'} does not match destination format of 'AWS::Lambda::Function.Arn'
template.yaml:40:9:40:22:E1041:{'Ref': 'MyAlias'} does not match destination format of 'AWS::Lambda::Function.Arn'
```
